pkg/bmh/list.go

```diff
@@ -0,0 +1,108 @@
+package bmh
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	bmhv1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+)
+
+const (
+	fiveScds time.Duration = 5 * time.Second
+)
+
+// List returns bareMetalHosts inventory in the given namespace.
+func List(apiClient *clients.Settings, nsname string) ([]*BmhBuilder, error) {
+	glog.V(100).Infof("Listing bareMetalHosts in the nsname %s", nsname)
+
+	if nsname == "" {
+		glog.V(100).Infof("bareMetalHost 'nsname' parameter can not be empty")
+
+		return nil, fmt.Errorf("failed to list bareMetalHosts, 'nsname' parameter is empty")
+	}
+
+	var bmhList bmhv1alpha1.BareMetalHostList
+	err := apiClient.List(context.Background(), &bmhList, &goclient.ListOptions{Namespace: nsname})
+
+	if err != nil {
+		glog.V(100).Infof("Failed to list bareMetalHosts in the nsname %s due to %s", nsname, err.Error())
+
+		return nil, err
+	}
+
+	if len(bmhList.Items) < 1 {
```

Comment:
This isn't required. Just let the function return an empty list and let the user determine if the cluster is in a bad state or not.